server/src/main/java/org/elasticsearch/search/aggregations/metrics/MultiValueAggregation.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.aggregations.metrics;
+
+import org.elasticsearch.search.aggregations.Aggregation;
+
+public interface MultiValueAggregation extends Aggregation {
+
+    /**
+     * Return an iterable over all value names this multi value aggregation provides.
+     *
+     * The iterable might be created on the fly, if you need to call this multiple times, please
+     * cache the result in a variable on caller side..
+     *
+     * @return iterable over all value names
+     */
+    Iterable<String> valueNames();
+
+    /**
+     * Return an iterable over all results by name
+     *
+     * @param name of the value
+     * @return iterable over all values formatted as string
+     */
+    Iterable<String> getValuesAsStrings(String name);
+
+    /**
+     * Return the maximum number of results per value
+     *
+     * Note: A single result might not have `size()` values
+     *
+     * @return maximum number of results per value
```

Comment:
Do you need this one? It feels fairly top-metrics specific.

Reply:
yes, I will need this for transform later. I basically want to know the size before iterating over the values. I could instead make `getValuesAsStrings` returning a `List`, but I like `Iterable` more, although it is a `List`.

Reply:
There is something against `size`: it basically says that all values have that size. That might not be true and/or limit us in future.

WDYT about removing and changing `getValuesAsStrings` to return `List`?

Reply:
> There is something against `size`: it basically says that all values have that size. That might not be true and/or limit us in future.

Yeah, that is the problem I had with `size`. We could make a `maxSize` if that's something that you'd need. But getting the results as a list seems sensible to me.